extensions-core/lookups-cached-global/src/main/java/io/druid/query/lookup/namespace/URIExtractionNamespace.java

```diff
@@ -185,45 +199,64 @@ public int hashCode()
     int result = getUri() != null ? getUri().hashCode() : 0;
     result = 31 * result + (getUriPrefix() != null ? getUriPrefix().hashCode() : 0);
     result = 31 * result + getNamespaceParseSpec().hashCode();
+    result = 31 * result + maps.hashCode();
     result = 31 * result + (getFileRegex() != null ? getFileRegex().hashCode() : 0);
     result = 31 * result + pollPeriod.hashCode();
     return result;
   }
 
-  private static class DelegateParser implements Parser<String, String>
+  public DelegateParser getParser(Parser<String, Object> delegate, String id)
+  {
+    return new DelegateParser(delegate, id);
+  }
+
+  private class DelegateParser implements Parser<MultiKey, Map<String, String>>
   {
     private final Parser<String, Object> delegate;
-    private final String key;
-    private final String value;
+    private final String id;
 
     private DelegateParser(
         Parser<String, Object> delegate,
-        @NotNull String key,
-        @NotNull String value
+        String id
     )
     {
       this.delegate = delegate;
-      this.key = key;
-      this.value = value;
+      this.id = id;
     }
 
     @Override
-    public Map<String, String> parse(String input)
+    public Map<MultiKey, Map<String, String>> parse(String input)
     {
       final Map<String, Object> inner = delegate.parse(input);
-      final String k = Preconditions.checkNotNull(
-          inner.get(key),
-          "Key column [%s] missing data in line [%s]",
-          key,
-          input
-      ).toString(); // Just in case is long
-      final String val = Preconditions.checkNotNull(
-          inner.get(value),
-          "Value column [%s] missing data in line [%s]",
-          value,
-          input
-      ).toString();
-      return ImmutableMap.<String, String>of(k, val);
+      ImmutableMap.Builder<MultiKey, Map<String, String>> builder = new ImmutableMap.Builder<>();
+
+      // somewhat dizzy to support simple json case
```

Comment:
See comment above… I wonder if it'd be less dizzy to move `"maps"` into the namespaceParseSpec.